src/aws/lambda-convert/src/convertClassroomRecording.js

```diff
@@ -23,3 +34,45 @@ module.exports = async (recordUrl, vodKey, sourceBucket) => {
     })
     .promise();
 };
+
+const getVideoStreamFromPlayback = async (redirectedResponse) => {
+  const cookies = redirectedResponse.headers.get('set-cookie') || '';
+  const rawCookie = cookies.split(';');
+  const locationURL = new URL(
+    redirectedResponse.headers.get('location').replace('http:', 'https:'),
+    redirectedResponse.url.replace('http:', 'https:'),
+  );
+  // Follow redirection to the playback page
+  const playbackPage = await fetch(locationURL, {
+    headers: {
+      Cookie: rawCookie[0],
+    },
+    redirect: 'manual',
+  });
+
+  // Follow redirection to the html video page
+  const videoLocationUrl = new URL(
+    playbackPage.headers.get('location').replace('http:', 'https:'),
+    playbackPage.url.replace('http:', 'https:'),
+  );
+  const videoPage = await fetch(videoLocationUrl, {
+    headers: {
+      Cookie: rawCookie[0],
+      Accept:
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
+      'Cache-Control': 'no-cache',
+    },
+    redirect: 'manual',
+  });
+  const baseUrl = videoPage.url.replace(/\/$/, '');
+  const { document } = new JSDOM(await videoPage.text()).window;
+  const source = document.querySelector('source');
```

Comment:
```suggestion
  const source = document.querySelector('source[type="video/mp4"]');
```
Maybe it would be more robust if the html playback changes ?

Reply:
I tried something like `video/*` if I remember but it doesn't work. I don't want to be block on the mp4 only

Reply:
ok, I see :+1: